src/Components/Search/NewSearch/NewSearchInputPillsContainer.tsx

```diff
@@ -0,0 +1,203 @@
+      if (pillsRef.current) {
+        const pillsContainer = pillsRef.current
+        if (pillsContainer.scrollWidth <= pillsContainer.clientWidth) {
+          setShowNextChevron(false)
+        } else {
+          setShowNextChevron(true)
+        }
+      }
+    }
+
+    window.addEventListener("resize", handleResize)
+    return () => window.removeEventListener("resize", handleResize)
+  }, [])
+
+  const scrollToLeft = () => {
+    if (pillsRef.current) {
+      const pillsContainer = pillsRef.current
+      const pills = pillsContainer.children
+      const currentScroll = pillsContainer.scrollLeft
+      const visibleWidth = pillsContainer.offsetWidth
+      let scrollBy = 0
+
+      // Find the last visible pill and the next pill
+      for (let i = pills.length - 1; i >= 0; i--) {
+        const pill = pills[i] as HTMLElement
+        if (
+          pill.offsetLeft + pill.offsetWidth <=
+          currentScroll + visibleWidth
+        ) {
+          const prevPill = pills[i - 1] as HTMLElement
+          scrollBy = -prevPill.offsetWidth
+          break
+        }
+      }
+
+      pillsContainer.scrollBy({
+        left: scrollBy - GRADIENT_BG_WIDTH,
+        behavior: "smooth",
+      })
+    }
+  }
+
+  const scrollToRight = () => {
+    if (pillsRef.current) {
+      const pillsContainer = pillsRef.current
+      const pills = pillsContainer.children
+      const currentScroll = pillsContainer.scrollLeft
+      let scrollBy = 0
+
+      // Find the first visible pill and the next pill
+      for (let i = 0; i < pills.length; i++) {
+        const pill = pills[i] as HTMLElement
+        if (pill.offsetLeft >= currentScroll) {
+          const nextPill = pills[i + 1] as HTMLElement
+          scrollBy = nextPill.offsetWidth
+          break
+        }
+      }
+
+      pillsContainer.scrollBy({
+        left: scrollBy + GRADIENT_BG_WIDTH,
+        behavior: "smooth",
+      })
+    }
+  }
+
+  const handleScroll = () => {
+    if (pillsRef.current) {
+      const currentPosition = pillsRef.current.scrollLeft
+      const maxScroll =
+        pillsRef.current.scrollWidth - pillsRef.current.clientWidth
+
+      // Check if the scroll position is at the beginning or end
+      if (currentPosition === 0) {
+        // Hide PreviousChevron when pills is at the beginning
+        setShowPreviousChevron(false)
+        setShowNextChevron(true)
+      } else if (currentPosition === maxScroll) {
+        // Hide NextChevron when pills reaches the end
+        setShowPreviousChevron(true)
+        setShowNextChevron(false)
+      } else {
+        // Show both chevrons otherwise
+        setShowPreviousChevron(true)
+        setShowNextChevron(true)
+      }
+    }
+  }
+
+  return (
+    <Flex alignItems="center" bg="white">
+      <Flex position="absolute" left={0} zIndex={showPreviousChevron ? 0 : -1}>
+        <PreviousChevron onClick={scrollToLeft} left={2} />
+        <GradientBg placement="left" />
+      </Flex>
```

Comment:
**question**: should we unmount, instead of changing `zIndex` when `showPreviousChevron` is `false`? Like this

```javascript
{showPreviousChevron && (
  <Flex position="absolute" left={0} zIndex={1}>
    <PreviousChevron onClick={scrollToLeft} left={2} />
    <GradientBg placement="left" />
  </Flex>
)}
```

Reply:
the same for the next chevron button

Reply:
With ZIndex I wrote less code so I think its good like this. But if you have a better reason to have mount/unmount logic we can replace it 👍🏽.

Reply:
I think then instead of changing `zIndex` (usually a negative value is not used in Force) it is better to use [this approach](https://github.com/artsy/palette/blob/ffa2c198375e7853516cae20d7f6002db48b4b5c/packages/palette/src/elements/Shelf/ShelfNavigation.tsx#L29-L30)